src/Raven.Server/Documents/DocumentDatabase.cs

```diff
@@ -416,6 +417,11 @@ private unsafe void DisposeInternal()
 
             var exceptionAggregator = new ExceptionAggregator(_logger, $"Could not dispose {nameof(DocumentDatabase)} {Name}");
 
+            var lockTaken = false;
+            Monitor.TryEnter(_locker, TimeSpan.FromSeconds(5), ref lockTaken);
+            if (lockTaken == false && _logger.IsOperationsEnabled)
```

Comment:
Let's wrap this in a `try/finally`, to be safe.

Reply:
I know that we are using the aggregator here, but it is possible that other things could throw that aren't expected and it bugs me that the lock may be leaked.

Reply:
Done